doc/heps/hep2.md

```diff
@@ -0,0 +1,372 @@
+  - [ ] **Must** ensure the warning was programmatically emitted at a level sufficient enough for users to see it.
+  - [ ] **Must** include the change in a major or minor release, not in a patch release.
+  - [ ] **Must** remove it from the code base and documentation.
+  - [ ] **Must** mention the removal in the release notes.
+  - [ ] **Must** close all open related issues and PRs.
+
+#### Test suite guidelines
+
+The Projects **must** be set up so that their test suite fails when they trigger deprecation warnings emitted by other HoloViz projects.
+
+## Appendix: Detour on Python warnings applied to HoloViz
+
+### Warning types
+
+Python defines the following [warning types](https://docs.python.org/3/library/warnings.html#warning-categories):
+
+| Class                     | Description                                                                                                                                                                   |
+| ------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `DeprecationWarning`        | Base category for warnings about deprecated features when those warnings are intended for other Python developers (ignored by default, unless triggered by code in `__main__`). |
+| `FutureWarning`             | Base category for warnings about deprecated features when those warnings are intended for end users of applications that are written in Python.                               |
+| `PendingDeprecationWarning` | Base category for warnings about features that will be deprecated in the future (ignored by default).                                                                         |
+
+The warnings defined to be used when deprecating a feature are `DeprecationWarning`, `FutureWarning`, and `PendingDeprecationWarning`.
+
+### Warnings filters and `stacklevel`
+
+Python provides the `warnings` module whose `warn` function is usually called by developers to issue warnings. These warnings can be filtered by users when running some code. The default warning filter has the following entries (in order of precedence):
+
+```
+default::DeprecationWarning:__main__
+ignore::DeprecationWarning
+ignore::PendingDeprecationWarning
+ignore::ImportWarning
+ignore::ResourceWarning
+```
+
+Note that the default filters can be overriden by setting the `-W` flag when calling the `python` executable or via the `PYTHONWARNINGS` environment variable.
+
+The default filters imply that in a regular context `PendingDeprecationWarning` will not be seen by the user, while `FutureWarning` will always be seen. `DeprecationWarning` is treated specially; warnings of these types are only seen when they are triggered by and points to code in `__main__`, the latter depending on the value of `stacklevel`.
```

Comment:
I can't parse "seen when they are triggered by and points to code in".

Reply:
Rephrased in https://github.com/holoviz/holoviz/pull/388/commits/c91e568e67be78b7025572e7dc503f9fbd851244

> `DeprecationWarning` is treated specially; warnings of these types are displayed only if `warn` is called by code located in the `__main__` namespace and if `warn` is configured to emit the warning in the `__main__` namespace, the latter depending on the value of `stacklevel`.